spec/generators/install_generator_spec.rb

```diff
@@ -20,68 +20,26 @@ class ApplicationController < Administrate::ApplicationController
     end
   end
 
-  describe "dashboard_manifest" do
-    it "is copied to the dashboards folder" do
-      stub_generator_dependencies
-      manifest = file("app/dashboards/dashboard_manifest.rb")
-
-      run_generator
-
-      expect(manifest).to exist
-      expect(manifest).to have_correct_syntax
-      expect(manifest).to contain("class DashboardManifest")
-      expect(manifest).to contain("DASHBOARDS = [")
-      expect(manifest).to contain("ROOT_DASHBOARD = DASHBOARDS.first")
-    end
-
-    it "populates default dashboards based on current ActiveRecord models" do
-      stub_generator_dependencies
-      manifest = file("app/dashboards/dashboard_manifest.rb")
-
-      run_generator
-
-      [:customers, :line_items, :orders, :products].each do |model|
-        expect(manifest).to contain(":#{model}")
-      end
-      expect(manifest).not_to contain("Delayed::Backend::ActiveRecord::Job")
-    end
-
-    it "skips namespaced models with a warning" do
-      stub_generator_dependencies
-      manifest = file("app/dashboards/dashboard_manifest.rb")
-
-      run_generator
-
-      expect(manifest).not_to contain("delayed/backend/active_record/jobs")
-    end
-
-    it "skips models that aren't backed by the database" do
+  describe "manifest generator" do
+    it "invokes the Manifest generator if the manifest does not yet exist" do
       begin
-        class ModelWithoutDBTable < ActiveRecord::Base; end
+        remove_constants :DashboardManifest
         stub_generator_dependencies
-        manifest = file("app/dashboards/dashboard_manifest.rb")
 
-        run_generator
+        expect { run_generator }.to raise_error
```

Comment:
This error is only raised in the test suite.

When a user runs this generator without `DashboardManifest` defined, the generator will invoke the manifest generator to fill in the missing piece.

In the test suite, we stub out the invocation of the manifest generator. The generator does not run, and `DashboardManifest` remains undefined. When the install generator tries to use `DashboardManifest` as the basis for the installation, it comes up empty and raises an error.